http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/LeoRoutesJsonCodec.scala

```diff
@@ -0,0 +1,295 @@
+          scopes.getOrElse(Set.empty),
+          enableWelder,
+          customClusterEnvironmentVariables.getOrElse(Map.empty)
+        )
+  }
+
+  implicit val runtimeConfigWriter: RootJsonWriter[RuntimeConfig] = (obj: RuntimeConfig) => {
+    val allFields = obj match {
+      case x: RuntimeConfig.GceConfig => Map(
+        "machineType" -> x.machineType.value.toJson,
+        "diskSize" -> x.diskSize.toJson,
+        "cloudService" -> x.cloudService.asString.toJson
+      )
+      case x: RuntimeConfig.DataprocConfig => Map(
+        "numberOfWorkers" -> x.numberOfWorkers.toJson,
+        "masterMachineType" -> x.masterMachineType.toJson,
+        "masterDiskSize" -> x.masterDiskSize.toJson,
+        "workerMachineType" -> x.workerMachineType.toJson,
+        "workerDiskSize" -> x.workerDiskSize.map(_.toJson).getOrElse(JsNull),
+        "cloudService" -> x.cloudService.asString.toJson,
+        "numberOfWorkerLocalSSDs" -> x.numberOfWorkerLocalSSDs.map(_.toJson).getOrElse(JsNull),
+        "numberOfPreemptibleWorkers" -> x.numberOfPreemptibleWorkers.map(_.toJson).getOrElse(JsNull)
+      )
+    }
+
+    val presentFields = allFields.filter(_._2 != JsNull)
+
+    JsObject(presentFields)
+  }
+
+  implicit val listClusterResponseWriter: RootJsonWriter[ListClusterResponse] = (obj: ListClusterResponse) => {
+    val allFields = Map(
+      "id" -> obj.id.toJson,
+      "internalId" -> obj.internalId.asString.toJson,
+      "clusterName" -> obj.clusterName.toJson,
+      "googleId" -> obj.dataprocInfo.map(_.googleId.toJson).getOrElse(JsNull),
+      "googleProject" -> obj.googleProject.toJson,
+      "serviceAccountInfo" -> obj.serviceAccountInfo.toJson,
+      "machineConfig" -> obj.machineConfig.toJson,
+      "clusterUrl" -> obj.clusterUrl.toJson,
+      "operationName" -> obj.dataprocInfo.map(_.operationName.toJson).getOrElse(JsNull),
+      "status" -> obj.status.toJson,
+      "hostIp" -> obj.dataprocInfo.map(_.hostIp.toJson).getOrElse(JsNull),
+      "creator" -> obj.auditInfo.creator.toJson,
+      "createdDate" -> obj.auditInfo.createdDate.toJson,
+      "destroyedDate" -> obj.auditInfo.destroyedDate.toJson,
+      "kernelFoundBusyDate" -> obj.auditInfo.kernelFoundBusyDate.toJson,
+      "labels" -> obj.labels.toJson,
+      "jupyterExtensionUri" -> obj.jupyterExtensionUri.toJson,
+      "jupyterUserScriptUri" -> obj.jupyterUserScriptUri.toJson,
+      "stagingBucket" -> obj.dataprocInfo.map(_.stagingBucket.toJson).getOrElse(JsNull),
+      "instances" -> obj.instances.toJson,
+      "dateAccessed" -> obj.auditInfo.dateAccessed.toJson,
+      "autopauseThreshold" -> obj.autopauseThreshold.toJson,
+      "defaultClientId" -> obj.defaultClientId.toJson,
+      "stopAfterCreation" -> obj.stopAfterCreation.toJson,
+      "welderEnabled" -> obj.welderEnabled.toJson,
+      "scopes" -> List
+        .empty[String]
+        .toJson //TODO: stubbing this out temporarily until AOU move to new swagger generated client
```

Comment:
What does this todo mean?

Reply:
Oh - perhaps we'd like to remove `scopes` in the listClusters response but AoU is still depending on it somehow

Reply:
Yea. It was previously stubbed out values from leo code. At some point, I moved the stubbing to decoder so that internally, things are cleaner. We might be able to remove this now, but I don't want to introduce api changes along with this PR